pandas/core/sorting.py

```diff
@@ -285,6 +317,30 @@ def nargsort(
     return indexer
 
 
+def ensure_key_mapped(values, key: Optional[Callable]):
+    """
+    Applies a callable key function to elements in an Index subclass or
+    an ndarray. Uses index.map for index subclasses and ignores nan values
+    in ndarrays.
+
+    Parameters
+    ----------
+    values : Index subclass or ndarray
+    key : Optional[Callable], key to be called on every index or entry in ndarray.
+    """
+    from pandas import Index
```

Comment:
can use ABCIndex here (and then can import from the top)

Reply:
`ABCIndexClass` works, but not `ABCIndex`. That fails on `RangeIndex`. Is that OK?